internal/r8/deobfuscator_test.go

```diff
@@ -18,41 +18,272 @@
 package r8
 
 import (
-	"fmt"
 	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/apm-data/model"
 )
 
-func TestDeobfuscation(t *testing.T) {
-	cases := []string{
-		"../../testdata/r8/deobfuscator/1",
-		"../../testdata/r8/deobfuscator/2",
-		"../../testdata/r8/deobfuscator/3",
-	}
+type FrameValidation struct {
+	updated           bool
+	originalClassname string
+	originalFunction  string
+	classname         string
+	function          string
+}
 
-	for _, c := range cases {
-		inputPath := c + "/obfuscated-crash"
-		expectedOutputPath := c + "/de-obfuscated-crash"
-		mapFilePath := c + "/mapping"
-
-		t.Run(fmt.Sprintf("(%s)->(%s)", inputPath, expectedOutputPath), func(t *testing.T) {
-			t.Parallel()
-			reader, err := os.Open(mapFilePath)
-			require.Nil(t, err)
-			defer reader.Close()
-			obfuscated := readFile(t, inputPath)
-			expected := readFile(t, expectedOutputPath)
-			deobfuscated, _ := Deobfuscate(obfuscated, reader)
-			assert.Equal(t, expected, deobfuscated)
-		})
-	}
+func TestSimpleLineDeobfuscation(t *testing.T) {
+	// Input:
+	// at androidx.appcompat.view.menu.e.f(Unknown Source:4)
+
+	// Expected output:
+	// at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(Unknown Source:4)
+	frame := createStacktraceFrame(4, "Unknown Source", "androidx.appcompat.view.menu.e", "f")
+	var stacktrace = new(model.Stacktrace)
+	*stacktrace = append(*stacktrace, frame)
+
+	mapFilePath := "../../testdata/r8/deobfuscator/1/mapping"
+	reader, err := os.Open(mapFilePath)
+	require.Nil(t, err)
```

Comment:
**suggestion**: I think it's better to use `require.NoError` for usability

Reply:
Thanks, it's updated now

Reply:
Thanks, it's updated now